src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs

```diff
@@ -154,125 +153,89 @@ public async Task<IReadOnlyCollection<JobInfo>> DequeueJobsAsync(byte queueType,
 
         public async Task<JobInfo> DequeueAsync(byte queueType, string worker, int heartbeatTimeoutSec, CancellationToken cancellationToken, long? jobId = null)
         {
-            try
-            {
-                using SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true);
-                using SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand();
-
-                // cannot use VLatest as it incorrectly asks for optional @InputJobId
-                sqlCommandWrapper.CommandText = "dbo.DequeueJob";
-                sqlCommandWrapper.CommandType = System.Data.CommandType.StoredProcedure;
-                sqlCommandWrapper.Parameters.AddWithValue("@QueueType", queueType);
-                sqlCommandWrapper.Parameters.AddWithValue("@Worker", worker);
-                sqlCommandWrapper.Parameters.AddWithValue("@HeartbeatTimeoutSec", heartbeatTimeoutSec);
-                if (jobId.HasValue)
+            return await _sqlRetryService.ExecuteSqlCommandFuncWithRetries(
```

Comment:
>ExecuteSqlCommandFuncWithRetries [](http://example.com/codeflow?start=42&length=32)

We should use ExecuteSqlReaderWithRetries here which should enforce correct usage of sql reader. 
Current code is incorrect as it does not do reader.NextResultAsync(cancellationToken). 
I added comment about this before. Please address.